src/MessagePack.UnityClient/Assets/Scripts/MessagePack/Formatters/StandardClassLibraryFormatter.cs

```diff
@@ -108,13 +108,78 @@ private DecimalFormatter()
 
         public void Serialize(ref MessagePackWriter writer, decimal value, MessagePackSerializerOptions options)
         {
-            writer.Write(value.ToString(CultureInfo.InvariantCulture));
-            return;
+            var dest = writer.GetSpan(MessagePackRange.MaxFixStringLength);
+            if (System.Buffers.Text.Utf8Formatter.TryFormat(value, dest.Slice(1), out var written))
+            {
+                // write header
+                dest[0] = (byte)(MessagePackCode.MinFixStr | written);
+                writer.Advance(written + 1);
+            }
+            else
+            {
+                // reset writer's span previously acquired that does not use
+                writer.Advance(0);
```

Comment:
FYI I researched this. Per the `IBufferWriter<T>` contract, this is a good change. Our `MessagePackWriter` happens to not require it. I would keep your `Advance(0)` call though, just as a good general pattern.